packages/x-codemod/src/v6.0.0/pickers/migrate-to-components-componentsProps/index.ts

```diff
@@ -0,0 +1,85 @@
+import { transformNestedProp } from '../../../util/addComponentsSlots';
+import removeProps from '../../../util/removeProps';
+
+const propsToSlots = {
+  // components
+  TransitionComponent: { prop: 'components', path: 'DesktopTransition' },
+
+  // componentsProps
+  PopperProps: { prop: 'componentsProps', path: 'popper' },
+  DialogProps: { prop: 'componentsProps', path: 'dialog' },
+  PaperProps: { prop: 'componentsProps', path: 'desktopPaper' },
+  TrapFocusProps: { prop: 'componentsProps', path: 'desktopTrapFocus' },
+  InputProps: { prop: 'componentsProps', path: 'input.InputProps' },
+  InputAdornmentProps: { prop: 'componentsProps', path: 'inputAdornment' },
+  OpenPickerButtonProps: { prop: 'componentsProps', path: 'openPickerButton' },
+};
+
+export default function transformer(file, api, options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+
+  const printOptions = options.printOptions || {
+    quote: 'single',
+    trailingComma: true,
+  };
+
+  root
+    .find(j.ImportDeclaration)
+    .filter(({ node }) => {
+      return node.source.value.startsWith('@mui/x-date-picker');
+    })
+
+    .forEach((path) => {
```

Comment:
Almost all of this transformation code could probably be re-used by other transformations doing the same or am I missing something? 🤔

Reply:
Yes, but I did not find a clear way to do it. The `transformNestedProp` already took a lot of time to have something flexible enough, so I just did some copy pasted